src/blocks/blocks.service.ts

```diff
@@ -156,6 +157,56 @@ export class BlocksService {
     );
   }
 
+  totalAndCirculatingSupplies(head: number): {
+    circulating: number;
+    total: number;
+  } {
+    let miningRewards = 0;
+    for (let sequence = 2; sequence <= head; sequence++) {
+      miningRewards += this.miningReward(sequence);
+    }
+
+    return {
+      circulating: this.circulatingSupply(head) + miningRewards,
+      total: GENESIS_SUPPLY_IN_IRON + miningRewards,
+    };
+  }
+
+  private circulatingSupply(sequence: number): number {
+    const monthsAfterLaunch = Math.floor(sequence / IRON_FISH_MONTH_IN_BLOCKS);
+
+    // Immediately available:
+    // 18.00% - Foundation
+    // 05.00% - IF Labs
+    // 02.25% - Airdrop
+    const immediatelyUnlocked = 0.2525;
+    let circulatingSupply = immediatelyUnlocked * GENESIS_SUPPLY_IN_IRON;
+
+    // Subject to 1y lock and 1y unlock:
+    // 05.10% - Preseed
+    // 09.90% - Seed
+    // 14.50% - Series A
+    // 00.60% - Advisors
+    // 37.40% - Team
+    // 05.00% - Future Endowments
+    const yearLockAndYearUnlock = 0.725;
+    if (monthsAfterLaunch >= 12) {
```

Comment:
Just to verify, this means half will be unlocked at 12 months, then the rest is slowly unlocked each month for the next year.

Reply:
Yeah